ts/screens/onboarding/EmailInsertScreen.tsx

```diff
@@ -0,0 +1,201 @@
+
+import { none, Option, some } from "fp-ts/lib/Option";
+import * as pot from "italia-ts-commons/lib/pot";
+import { EmailString } from "italia-ts-commons/lib/strings";
+import { untag } from "italia-ts-commons/lib/types";
+
+import { Content, Form, Text, View } from "native-base";
+
+import * as React from "react";
+
+import {
+  Alert,
+  KeyboardAvoidingView,
+  Platform,
+  StyleSheet
+} from "react-native";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import { connect } from "react-redux";
+import { LabelledItem } from "../../components/LabelledItem";
+import BaseScreenComponent from "../../components/screens/BaseScreenComponent";
+import FooterWithButtons from "../../components/ui/FooterWithButtons";
+import H4 from "../../components/ui/H4";
+import Markdown from "../../components/ui/Markdown";
+import I18n from "../../i18n";
+import { abortOnboarding, emailInsert } from "../../store/actions/onboarding";
+import { ReduxProps } from "../../store/actions/types";
+import { GlobalState } from "../../store/reducers/types";
+import customVariables from "../../theme/variables";
+
+type OwnProps = {
+  navigation: NavigationScreenProp<NavigationState>;
+};
+
+type Props = ReduxProps & OwnProps & ReturnType<typeof mapStateToProps>;
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1
+  },
+  content: {
+    flex: 1
+  },
+  horizontalPadding: {
+    paddingHorizontal: customVariables.contentPadding
+  },
+  boldH4: {
+    fontWeight: customVariables.textBoldWeight,
+    paddingTop: customVariables.spacerLargeHeight
+  },
+  icon: {
+    marginTop: Platform.OS === "android" ? 4 : 6 // correct icon position to align it with baseline of email text}
+  },
+  emailInput: {
+    fontWeight: customVariables.h1FontWeight,
+    color: customVariables.h1Color,
+    fontSize: 18
+  }
+});
+
+const EMPTY_EMAIL = "";
+
+type State = Readonly<{
+  email: Option<string>;
+}>;
+
+/**
+ * A screen to allow user to insert an email address.
+ */
+class EmailInsertScreen extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { email: this.props.email };
+  }
+
+  /**
+   * Footer
+   */
+  private renderFooterButtons() {
+    const continueButtonProps = {
+      disabled: this.isValidEmail() !== true,
+      onPress: () => this.props.dispatch(emailInsert()),
+      title: I18n.t("global.buttons.continue"),
+      block: true,
+      primary: this.isValidEmail()
+    };
+
+    return (
+      <FooterWithButtons type="SingleButton" leftButton={continueButtonProps} />
+    );
+  }
+
+  /** validate email returning three possible values:
+   * - _true_,      if email is valid.
+   * - _false_,     if email has been already changed from the user and it is not
+   * valid.
+   * - _undefined_, if email field is empty. This state is consumed by
+   * LabelledItem Component and it used for style pourposes ONLY.
+   */
+  private isValidEmail() {
+    return this.state.email
```

Comment:
```suggestion
    return this.state.email.fold(false, value =>
      EmailString.decode(value).isRight()
    );
```

Reply:
It's needed a 3-way value ( boolean | undefined). I see that fold returns only a boolean.